plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java

```diff
@@ -261,6 +268,38 @@ public Transaction newCreateTableTransaction(
                 isUsingSystemSecurity ? Optional.empty() : Optional.of(session.getUser()));
     }
 
+    @Override
+    public void registerTable(ConnectorSession session, SchemaTableName schemaTableName, String metadataLocation)
+            throws TrinoException
+    {
+        if (tableExists(schemaTableName)) {
+            throw new TrinoException(TABLE_ALREADY_EXISTS, format("Table '%s' already exists", schemaTableName));
+        }
+        Optional<String> owner = isUsingSystemSecurity ? Optional.empty() : Optional.of(session.getUser());
+
+        String tableLocation = metadataLocation.replaceFirst("/metadata/.*", "");
```

Comment:
Rather than this, it would be better to have the caller pass it explicitly.
Why Glue doesn't need this and HMS catalog doeS?

Reply:
Modified the method signature to pass `tableLocation`